src/test/java/games/strategy/util/VersionTest.java

```diff
@@ -12,66 +13,70 @@
   public void testCompare() {
     final Version v1 = new Version(0, 0, 0);
     final Version v2 = new Version(1, 0, 0);
-    assertFalse(v1.equals(v2));
-    assertFalse(v2.equals(v1));
+    assertNotEquals(v1, v2);
+    assertNotEquals(v2, v1);
   }
 
   @Test
   public void testCompare2() {
     final Version v1 = new Version(0, 0, 0);
     final Version v2 = new Version(1, 1, 0);
-    assertTrue(!v1.equals(v2));
-    assertTrue(!v2.equals(v1));
+    assertNotEquals(v1, v2);
+    assertNotEquals(v2, v1);
   }
 
   @Test
   public void testCompare3() {
     final Version v1 = new Version(0, 0, 0);
     final Version v2 = new Version(0, 1, 0);
-    assertTrue(!v1.equals(v2));
-    assertTrue(!v2.equals(v1));
+    assertNotEquals(v1, v2);
+    assertNotEquals(v2, v1);
   }
 
   @Test
   public void testCompare4() {
     final Version v1 = new Version(0, 0, 0);
     final Version v2 = new Version(0, 0, 1);
-    assertTrue(!v1.equals(v2));
-    assertTrue(!v2.equals(v1));
-  }
-
-  @Test
-  public void testCompare5() {
-    // micro differences should have no difference
-    final Version v1 = new Version(0, 0, 0, 0);
-    final Version v2 = new Version(0, 0, 0, 1);
-    assertTrue(v1.equals(v2, true));
-    assertTrue(v2.equals(v1, true));
+    assertNotEquals(v1, v2);
+    assertNotEquals(v2, v1);
   }
 
   @Test
   public void testRead1() {
-    assertTrue("1.2.3".equals(new Version("1.2.3").toString()));
+    assertEquals("1.2.3", new Version("1.2.3").toString());
   }
 
   @Test
   public void testRead2() {
-    assertTrue("1.2".equals(new Version("1.2").toString()));
+    assertEquals("1.2", new Version("1.2").toString());
   }
 
   @Test
   public void testRead3() {
-    assertTrue("1.2".equals(new Version("1.2.0").toString()));
+    assertEquals("1.2", new Version("1.2.0").toString());
   }
 
   @Test
   public void testRead4() {
-    assertTrue("1.2.3.dev".equals(new Version("1.2.3.dev").toString()));
+    assertEquals("1.2.3.dev", new Version("1.2.3.dev").toString());
   }
 
   @Test
   public void getExactVersion() {
     assertEquals("1.2.3.4", new Version(1, 2, 3, 4).getExactVersion());
     assertEquals("1.2.3.4.5", new Version("1.2.3.4.5").getExactVersion());
   }
+
+  @Test
+  public void testCompatible() {
+    assertTrue(new Version(1, 9).isCompatible(new Version(1, 9)));
```

Comment:
- recommend ordering the test values, this is something that is useful in combinatorics when you want to carefully enumerate all permutations, eg:
```
1.9 + 1.9
1.9 + 1.9.0
1.9 + 1.9.1
1.9 + 1.9.9
1.9 + 1.9.0.0
1.9 + 1.9.0.1
1.9 + 1.9.0.9
```

Second, Version can be serialized, if there is not a match then the fields or object itself may be null, so another interesting test case would be `null`